src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateGeneratorTests.cs

```diff
@@ -35,75 +37,58 @@ public async Task MapAction_NoParam_StringReturn(string source, string httpMetho
         await VerifyResponseBodyAsync(httpContext, expectedBody);
     }
 
-    [Fact]
-    public async Task MapAction_SingleStringParam_StringReturn()
-    {
-        var (results, compilation) = await RunGeneratorAsync("""
-app.MapGet("/hello", ([FromQuery]string p) => p);
-""");
-
-        var endpointModel = GetStaticEndpoint(results, GeneratorSteps.EndpointModelStep);
-        var endpoint = GetEndpointFromCompilation(compilation);
-
-        Assert.Equal("/hello", endpointModel.RoutePattern);
-        Assert.Equal("MapGet", endpointModel.HttpMethod);
-        var p = Assert.Single(endpointModel.Parameters);
-        Assert.Equal(EndpointParameterSource.Query, p.Source);
-        Assert.Equal("p", p.Name);
-
-        var httpContext = CreateHttpContext();
-        httpContext.Request.QueryString = new QueryString("?p=Hello%20world!");
-
-        await endpoint.RequestDelegate(httpContext);
-        await VerifyResponseBodyAsync(httpContext, "Hello world!");
-        await VerifyAgainstBaselineUsingFile(compilation);
-    }
-
-    [Fact]
-    public async Task MapAction_SingleNullableStringParam_WithQueryStringValueProvided_StringReturn()
+    public static object[][] MapAction_ExplicitQueryParam_StringReturn_Data
     {
-        var (results, compilation) = await RunGeneratorAsync("""
-app.MapGet("/hello", ([FromQuery]string? p) => p ?? "Error!");
-""");
-
-        var endpointModel = GetStaticEndpoint(results, GeneratorSteps.EndpointModelStep);
-        var endpoint = GetEndpointFromCompilation(compilation);
-
-        Assert.Equal("/hello", endpointModel.RoutePattern);
-        Assert.Equal("MapGet", endpointModel.HttpMethod);
-        var p = Assert.Single(endpointModel.Parameters);
-        Assert.Equal(EndpointParameterSource.Query, p.Source);
-        Assert.Equal("p", p.Name);
-
-        var httpContext = CreateHttpContext();
-        httpContext.Request.QueryString = new QueryString("?p=Hello%20world!");
+        get
+        {
+            var expectedBody = "TestQueryValue";
+            var fromQueryRequiredSource = $"""app.MapGet("/", ([{typeof(FromQueryAttribute)}] string queryValue) => queryValue);""";
```

Comment:
Do we really need to use `typeof` here? I think it is clutter and confusing. It isn't like we are ever going to rename `FromQueryAttribute`.

```suggestion
            var fromQueryRequiredSource = """app.MapGet("/", ([FromQuery] string queryValue) => queryValue);""";
```

Reply:
Although we need this for the `Todo` type as mentioned for the reasons below, we don't strictly need it for `FromQuery` type since that's only defined in MVC. The main benefit we gain here is not having to have the `using` for the attributes in the test source.